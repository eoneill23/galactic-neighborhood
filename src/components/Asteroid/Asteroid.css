.asteroid-container {
  border-bottom: 1px solid gray;
}

.name-span {
  font-weight: bold;
}

.name {
  color: yellow;
}

.hazardous {
  color: red;
}

.non-hazardous {
  color: green;
}

