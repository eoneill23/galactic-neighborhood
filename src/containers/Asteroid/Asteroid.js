import React from 'react';

const Asteroid = ({}) => {

  return (
    <>
    </>
  )
}

export default Asteroid;