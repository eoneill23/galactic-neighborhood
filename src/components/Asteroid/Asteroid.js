import React from 'react';
import './Asteroid.css';
import PropTypes from 'prop-types';

const Asteroid = ({ name, minDiam, maxDiam, isHazardous }) => {

  let hazardStatus = isHazardous ? 'It might be time to book that trip to Europe...' : 'No.'

  let hazardClassName = isHazardous ? 'hazardous' : 'non-hazardous'

  return (
    <article className='asteroid-container'>
      <p><span className='name-span'>NAME:</span> <span className='name'>{name}</span></p>
      <p><span className='name'>{name}</span> is estimated to be between {minDiam} & {maxDiam} miles in diameter.</p>
      <p>Is <span className='name'>{name}</span> hazardous? <span className={hazardClassName}>{hazardStatus}</span></p>
    </article>
  )
}

export default Asteroid;

Asteroid.propTypes = {
  name: PropTypes.string,
  minDiam: PropTypes.string,
  maxDiam: PropTypes.string,
  isHazardous: PropTypes.bool
}