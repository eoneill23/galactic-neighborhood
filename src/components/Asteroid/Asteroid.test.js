import React from 'react';
import { shallow } from 'enzyme';
import Asteroid from './Asteroid';

describe('Asteroid', () => {
  it('should match the snapshot with isHazardous set to false', () => {
    const falseWrapper = shallow(<Asteroid 
      name={'Big boy'}
      minDiam={'.002'}
      maxDiam={'.05'}
      isHazardous={false}
    />);

    expect(falseWrapper).toMatchSnapshot();
  });

  it('should match the snapshot with isHazardous set to true', () => {
    const trueWrapper = shallow(<Asteroid
      name={'Medium boy'}
      minDiam={'.003'}
      maxDiam={'.07'}
      isHazardous={true}
    />);

    expect(trueWrapper).toMatchSnapshot();
  });
});