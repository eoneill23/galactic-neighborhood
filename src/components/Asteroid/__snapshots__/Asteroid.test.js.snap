// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asteroid should match the snapshot with isHazardous set to false 1`] = `
<article
  className="asteroid-container"
>
  <p>
    <span
      className="name-span"
    >
      NAME:
    </span>
     
    <span
      className="name"
    >
      Big boy
    </span>
  </p>
  <p>
    <span
      className="name"
    >
      Big boy
    </span>
     is estimated to be between 
    .002
     & 
    .05
     miles in diameter.
  </p>
  <p>
    Is 
    <span
      className="name"
    >
      Big boy
    </span>
     hazardous? 
    <span
      className="non-hazardous"
    >
      No.
    </span>
  </p>
</article>
`;

exports[`Asteroid should match the snapshot with isHazardous set to true 1`] = `
<article
  className="asteroid-container"
>
  <p>
    <span
      className="name-span"
    >
      NAME:
    </span>
     
    <span
      className="name"
    >
      Medium boy
    </span>
  </p>
  <p>
    <span
      className="name"
    >
      Medium boy
    </span>
     is estimated to be between 
    .003
     & 
    .07
     miles in diameter.
  </p>
  <p>
    Is 
    <span
      className="name"
    >
      Medium boy
    </span>
     hazardous? 
    <span
      className="hazardous"
    >
      It might be time to book that trip to Europe...
    </span>
  </p>
</article>
`;
